src/http/server/server.cr

```diff
@@ -2,6 +2,20 @@ require "openssl"
 require "socket"
 require "../common"
 
+# A handler is a class which inherits from HTTP::Handler and implements the `call`method.
+# You can use a handler to intercept any incoming request and do something like request throttling, ip-based whitelisting e.g
```

Comment:
Practically, yes it is. I'm also curious why don't we just call them `Middleware` s